src/event_handler.rs

```diff
@@ -192,22 +188,125 @@ async fn send_execution_output(app_state: Data<AppState>, mut rx: Receiver<JobRe
+            let txn = job_txn.clone();
+
+            // Estimate gas required for the transaction to execute and retry otherwise
+            let estimated_gas = txn.estimate_gas().await;
+            let Ok(estimated_gas) = estimated_gas else {
+                eprintln!("Failed to estimate gas from the rpc for sending execution {} transaction for job id {}: {:?}", txn_type, job_id, estimated_gas.unwrap_err());
+                sleep(Duration::from_millis(10)).await;
+                continue;
+            };
+
+            // Get current gas price for the common chain network and retry otherwise
+            let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
+            let gas_price = http_rpc_client.get_gas_price().await;
+            let Ok(gas_price) = gas_price else {
+                eprintln!(
+                    "Failed to get gas price from the rpc for the network: {:?}",
+                    gas_price.unwrap_err()
                 );
+                sleep(Duration::from_millis(10)).await;
+                continue;
+            };
 
-                send_txn(txn).await
-            },
-            should_retry,
-        )
-        .await;
+            // If required retrieve the current nonce from the network and retry otherwise
+            if update_nonce == true {
+                let new_nonce_to_send = http_rpc_client
+                    .get_transaction_count(http_rpc_client.address(), None)
+                    .await;
+                if new_nonce_to_send.is_err() {
+                    eprintln!(
+                        "Failed to fetch current nonce for the gas address ({:?}): {:?}",
+                        http_rpc_client.address(),
+                        new_nonce_to_send.unwrap_err()
+                    );
+                    continue;
+                };
+                nonce_to_send = new_nonce_to_send.unwrap();
+                update_nonce = false;
+            }
 
-        let Ok(_) = txn_result else {
-            eprintln!(
-                "Failed to submit the execution output transaction for job id {}: {:?}",
-                job_output.id,
-                txn_result.unwrap_err()
+            // Update metadata to be used for sending the transaction and send it to the common chain
+            let txn = txn
+                .gas(estimated_gas + GAS_LIMIT_BUFFER)
+                .nonce(nonce_to_send)
+                .gas_price(gas_price);
+            let pending_txn = txn.send().await;
+            let Ok(pending_txn) = pending_txn else {
+                let error_string = format!("{:?}", pending_txn.unwrap_err());
+                eprintln!(
+                    "Failed to send the execution {} transaction for job id {}: {}",
+                    txn_type, job_id, error_string
+                );
+
+                // Retry if 'nonce too low' error encountered
+                if error_string.contains("code: -32000") && error_string.contains("nonce") {
```

Comment:
there are two kind of nonce error, nonce too low and too high. Keep that in mind while retrying.